app/src/main/java/com/example/weatherforecastapp/ui/activities/ForecastActivity.kt

```diff
@@ -0,0 +1,305 @@
+                    R.string.min_max_temp_fahr,
+                    data.forecast.week_forecast.get(0).day.min_temp_fahr.roundToInt(),
+                    data.forecast.week_forecast.get(0).day.max_temp_fahr.roundToInt()
+                )
+                binding.visibility.text =
+                    getString(R.string.visibility_mil, data.Current.visibility_miles.roundToInt())
+                binding.wind.text = getString(R.string.wind_mph, data.Current.wind_miles.roundToInt())
+                binding.visibility.text =
+                    getString(R.string.visibility_mil, data.Current.visibility_miles.roundToInt())
+            }
+
+
+            binding.humidity.text = getString(R.string.humidity, data.Current.humidity)
+            binding.pressure.text = getString(R.string.pressure, data.Current.pressure.roundToInt())
+
+
+
+            //DAY WEATHER
+            val todayView = binding.todayWeatherView
+            todayView.findViewById<TextView>(R.id.custom_title).text = getString(R.string.today)
+            val rvTodayWeathers = todayView.findViewById<RecyclerView>(R.id.recyclerView)
+            val todayWeathers = it.forecast.week_forecast[0].hour_forecast
+            val adapter1 = WeatherRecyclerAdapter(this, todayWeathers)
+            rvTodayWeathers.adapter = adapter1
+            rvTodayWeathers.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
+            val dividerItemDecoration = DividerItemDecoration(
+                rvTodayWeathers.context,
+                LinearLayoutManager.HORIZONTAL
+            )
+            rvTodayWeathers.addItemDecoration(dividerItemDecoration)
+
+            //WEEK WEATHER
+            val weekView = binding.weekWeatherView
+            weekView.findViewById<TextView>(R.id.custom_title).text = getString(R.string.next_seven_days)
+            val rvWeekWeathers = weekView.findViewById<RecyclerView>(R.id.recyclerView)
+            val weekWeathers = it.forecast.week_forecast
+            //Ovo je da maknemo danasnji dan
+            weekWeathers.removeFirst()
+            val adapter2 = WeatherRecyclerAdapter(this,weekWeathers)
+            // Attach the adapter to the recyclerview to populate items
+            rvWeekWeathers.adapter = adapter2
+            // Set layout manager to position the items
+            rvWeekWeathers.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
+            // That's all!
+            rvWeekWeathers.addItemDecoration(dividerItemDecoration)
+
+
+            val mapFragment = supportFragmentManager.findFragmentById(R.id.map_fragment) as SupportMapFragment?
+
+
+            mapFragment?.getMapAsync{googleMap ->
+                val cityName = data.Location.city_name
+
+
+                val geocoder = Geocoder(this, Locale.getDefault())
+                var lat: Double = 1.0
+                var long: Double = 1.0
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```

Comment:
default vrijednosti za long i lat su 1.0 sto znaci da je razlog zato uvijek dobivas te vrijednosti ili zato sto `getFromLocationName` faila pa se pozove `onError` metoda, ili zato sto ti je verzija OS-a manja od 33 (tiramisu) pa ti faila provjera na liniji 217. Ali, cak i da ti to dvoje oboje prodje uspjesno, treci razlog zbog kojeg ces dobiti defaultne vrijednosti je sto je `getFromLocationName` asinkrona metoda. Sto znaci da aplikacija na glavnoj dretvi izvodi slijedno jednu po jednu naredbu na liniji 214, 215, 217 i 218 te zatim skace na liniju 237 i crta mapu s defaultnim koordinatama. Za to vrijeme se na nekoj drugoj dretvi dohvacaju koordinate i kada one stignu, vec je prekasno jer se mapa vec iscrtala se defaultnim vrijednostima koordinata.